test/core.jl

```diff
@@ -1,3 +1,4 @@
+# FIXME: type stability broken. The following should NOT error
 A = @inferred(AxisArray(reshape(1:24, 2,3,4), .1:.1:.2, .1:.1:.3, .1:.1:.4))
```

Comment:
The type instability comes from `default_axes` which i tried to fix, but couldn't

Its quite strange that the type instability usually happens for the last tuple element

```
julia> @code_warntype AxisArrays.default_axes(reshape(1:24, 2,3,4), (.1:.1:.2, .1:.1:.3, .1:.1:.4))
Body::Tuple{Axis,Axis,Axis}
172 1 ─ %1 = (getfield)(args, 2)::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}                       │╻╷   _default_axes
    │   %2 = (getfield)(args, 3)::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}                       ││┃    tail
    │   %3 = (Base.getfield)(args, 1, true)::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}            ││╻    getindex
    │   %4 = %new(Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}, %3)::Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}
    │   %5 = (Core.tuple)(%2)::Tuple{StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}                   │││╻    tail
    │   %6 = %new(Axis{:col,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}, %1)::Axis{:col,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}
    │   %7 = (Core.tuple)(%4, %6)::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}}
    │   %8 = invoke AxisArrays._default_axes(_2::Base.ReshapedArray{Int64,3,UnitRange{Int64},Tuple{}}, %5::Tuple{StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}, %7::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}})::Tuple{Axis,Axis,Axis}
    └──      return %8 
```